assets/javascripts/discourse/initializers/hook-uploads.js.es6

```diff
@@ -0,0 +1,23 @@
+import { withPluginApi } from "discourse/lib/plugin-api";
+
+export default {
+  name: "hook-uploads",
+
+  initialize(container) {
+    const siteSettings = container.lookup("site-settings:main");
+    if (!siteSettings.encrypt_enabled) {
+      return;
+    }
+
+    withPluginApi("0.8.27", api => {
+      api.addComposerUploadHandler([".*"], () => {
+        const controller = container.lookup("controller:composer");
```

Comment:
This is the correct way to implement this, but looking over the code I realize a small change in core could make this a lot easier. What if the [composer upload handler](https://github.com/discourse/discourse/blob/master/app/assets/javascripts/discourse/components/composer-editor.js.es6#L684) called the handler with a reference to the controller as the second parameter?

Then your code could avoid the lookup. It seems like having a reference to the controller would be useful for most plugins and this API.